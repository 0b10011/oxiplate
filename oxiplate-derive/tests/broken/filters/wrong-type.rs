use oxiplate_derive::Oxiplate;

mod filters_for_oxiplate {
    use std::fmt::Display;

    pub fn respond(expression: impl Display) -> impl Display {
        let expression = expression.to_string();
        match expression.as_str() {
            "hello" => "world".to_string(),
            _ => "did not understand: ".to_string() + &expression,
        }
    }

    pub fn respond_string(expression: String) -> impl Display {
        respond(expression)
    }
}

struct Message(String);

#[derive(Oxiplate)]
#[oxiplate_inline(r#"{{ message | respond() }} {{ message | respond_string() }}"#)]
struct Data {
    message: Message,
}

fn main() {
    assert_eq!(format!("{}", Data { message: Message("goodbye".to_string()) }), "did not understand: goodbye");
}