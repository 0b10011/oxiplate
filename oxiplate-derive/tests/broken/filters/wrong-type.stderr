error[E0277]: `Message` doesn't implement `std::fmt::Display`
  --> tests/broken/filters/wrong-type.rs:22:25
   |
22 | #[oxiplate_inline(r#"{{ message | respond() }} {{ message | respond_string() }}"#)]
   |                         ^^^^^^^   ------- required by a bound introduced by this call
   |                         |
   |                         the trait `std::fmt::Display` is not implemented for `Message`
   |
note: required by a bound in `respond`
  --> tests/broken/filters/wrong-type.rs:6:37
   |
 6 |     pub fn respond(expression: impl Display) -> impl Display {
   |                                     ^^^^^^^ required by this bound in `respond`

error[E0277]: `Message` doesn't implement `std::fmt::Display`
  --> tests/broken/filters/wrong-type.rs:22:35
   |
22 | #[oxiplate_inline(r#"{{ message | respond() }} {{ message | respond_string() }}"#)]
   |                                   ^^^^^^^^^ the trait `std::fmt::Display` is not implemented for `Message`
   |
note: required by a bound in `respond`
  --> tests/broken/filters/wrong-type.rs:6:37
   |
 6 |     pub fn respond(expression: impl Display) -> impl Display {
   |                                     ^^^^^^^ required by this bound in `respond`

error[E0308]: mismatched types
  --> tests/broken/filters/wrong-type.rs:22:51
   |
22 | #[oxiplate_inline(r#"{{ message | respond() }} {{ message | respond_string() }}"#)]
   |                                                   ^^^^^^^   -------------- arguments to this function are incorrect
   |                                                   |
   |                                                   expected `String`, found `Message`
   |
note: function defined here
  --> tests/broken/filters/wrong-type.rs:14:12
   |
14 |     pub fn respond_string(expression: String) -> impl Display {
   |            ^^^^^^^^^^^^^^ ------------------
